packages/puppeteer-core/src/bidi/BrowserConnector.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2023 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {_getCdpConnection} from '../cdp/BrowserConnector.js';
+import type {ConnectionTransport} from '../common/ConnectionTransport.js';
+import type {
+  BrowserConnectOptions,
+  ConnectOptions,
+} from '../common/ConnectOptions.js';
+import {UnsupportedOperation} from '../common/Errors.js';
+import {debugError} from '../common/util.js';
+import {isNode} from '../environment.js';
+
+import type {BidiBrowser} from './Browser.js';
+import type {BidiConnection} from './Connection.js';
+
+const DEFAULT_VIEWPORT = Object.freeze({width: 800, height: 600});
```

Comment:
`DEFAULT_VIEWPORT` and `getWebSocketTransportClass` should be extracted into common if possible.
The more important for me is `DEFAULT_VIEWPORT` as it may not brake things if changed.

Reply:
Referencing common from cdp/bidi would create a cyclic references. We can extend `_connectToBiDiBrowser` with an additional parameter, but that would be awkward as well. Any ideas on how to address it?

Reply:
I don't think it will, we can move it to `../common/util.js` similar to what we had done for `NETWORK_IDLE_TIME`.

Reply:
got it. Done.